homeassistant/components/saj/sensor.py

```diff
@@ -91,9 +91,10 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         raise PlatformNotReady
 
     for sensor in sensor_def:
-        hass_sensors.append(
-            SAJsensor(saj.serialnumber, sensor, inverter_name=config.get(CONF_NAME))
-        )
+        if sensor.enabled:
+            hass_sensors.append(
+                SAJsensor(saj.serialnumber, sensor, inverter_name=config.get(CONF_NAME))
+            )
 
     async_add_entities(hass_sensors)
```

Comment:
```python
async_add_entities(
    [
        SAJsensor(saj.serialnumber, sensor, inverter_name=config.get(CONF_NAME))
        for sensor in sensor_def
        if sensor.enabled
    ]
)
```

Reply:
@bachya Looks nice but I use "hass_sensors" in async_saj() :
https://github.com/home-assistant/core/blob/c0e39b3c0b0498aae77ee239c1eadff3764e7f43/homeassistant/components/saj/sensor.py#L105

Can I request the added entities somehow or do I still need to store them in a separate list to use them later?

Reply:
Ah, okay — disregard. 👍🏻